src/parser/flow_lexer.ml

```diff
@@ -1613,16 +1751,39 @@ let type_token env lexbuf =
       Token (env, mk_num_singleton LEGACY_OCTAL num)
     | _ -> failwith "unreachable"
     )
-
   | Opt neg, legacyoctnumber ->
     let num = lexeme lexbuf in
     Token (env, mk_num_singleton LEGACY_OCTAL num)
 
+  | Opt neg, hexbigint, word ->
+    (* Numbers cannot be immediately followed by words *)
+    recover env lexbuf ~f:(fun env lexbuf -> match%sedlex lexbuf with
+    | Opt neg, hexbigint ->
+      let num = lexeme lexbuf in
+      (* Reconsider Windows support of FloatOfString *)
+      begin try Token (env, mk_bignum_singleton NORMAL num)
+      with _ when Sys.win32 ->
+        let loc = loc_of_lexbuf env lexbuf in
+        let env = lex_error env loc Parse_error.WindowsFloatOfString in
+        Token (env, T_BIGINT_SINGLETON_TYPE { kind = NORMAL; approx_value = 789.0; raw = "789" })
```

Comment:
Nit - I think `raw` should be a valid bigint, so it should probably end with `n`. Extra points to making it an authentic `hexbigint`. So maybe 

```
Token (env, T_BIGINT_SINGLETON_TYPE { kind = NORMAL; approx_value = 291.0; raw = "0x123n" })
```